.github/workflows/nox.yml

```
@@ -0,0 +1,34 @@
+---
+name: nox
+
+"on":
+  push:
+    branches-ignore:
+      - 'patchback/**'
+  pull_request:
+  workflow_dispatch:
```

Instead of copy-pasting triggers and having to keep the ignores and everything else in sync, change this to only be `workflow_call:` and rename the file to have a `reusable-` prefix.
This allows wiring everything into a single workflow runtime with the event triggers synchronized, the possibility to dynamically skip jobs and lets us use the alls-green gate.

I've been experimenting with this structural approach lately and successfully deployed in across some PyPA repos and CPython.

It's being picked up by some other subcommunities. For example, Henry documented it here: https://learn.scientific-python.org/development/guides/gha-basic/#reusable-workflows.

This way, the job definitions can remain modularized, in a dedicated YAML file, while allowing for complex job dependencies to be transparent in the main `ci-cd.yml`.

---

That seems more complicated than what we need here and better suited for a follow up PR.

---

Well, you're already copy-pasting this entire section. The patch would reduce the number of lines, making it simpler. I'm not asking to implement other aspects of the structural approach, just inclusion. Here, it'd be
```suggestion
  workflow_call:
```
And in the main workflow, it'd have a `uses:` job (can be 1 line).

---

I took a stab at this in https://github.com/ansible/ansible-documentation/pull/258/commits/a318c9d9f58d975474f88181a3ca6cfbe31a68de. Let's see if i did it correctly :).

---

@gotmax23 that's good, but could you also rename `nox.yml` to `reusable-nox.yml`. I use this convention to visually separate modules that are only intended to be reusable and never triggered standalone.